calico/etcddriver/driver.py

```diff
@@ -700,15 +741,25 @@ def watch_etcd(self, next_index, event_queue, stop_event):
                     _log.info("No HTTP pool, creating one...")
                     http = self.get_etcd_connection()
                 req_start_time = monotonic_time()
+                if req_end_time is not None:
+                    non_req_time = req_start_time - req_end_time
```

Comment:
No, but confusing!  I've added a comment.  We're calculating the time since the end of the last request so start_time is new and end_time was from the previous loop.